vue-app/src/components/TimeLeft.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <span
+    v-if="
+      timeLeft.days === 0 &&
+      timeLeft.hours === 0 &&
+      timeLeft.minutes === 0 &&
+      timeLeft.seconds === 0
+    "
+    :class="valueClass || 'value'"
+  >
+    {{ timeLeft.days }}
+    <span :class="unitClass || 'unit'"
+      >day{{ timeLeft.days !== 1 ? 's' : '' }}</span
+    >
+    {{ timeLeft.hours }}
+    <span :class="unitClass || 'unit'"
+      >hour{{ timeLeft.hours !== 1 ? 's' : '' }}</span
+    >
+  </span>
+  <span v-else-if="timeLeft.days > 0" :class="valueClass || 'value'">
+    {{ timeLeft.days }}
+    <span :class="unitClass || 'unit'"
+      >day{{ timeLeft.days !== 1 ? 's' : '' }}</span
+    >
+    {{ timeLeft.hours }}
+    <span :class="unitClass || 'unit'"
+      >hour{{ timeLeft.hours !== 1 ? 's' : '' }}</span
+    >
+  </span>
+  <span v-else-if="timeLeft.hours > 0" :class="valueClass || 'value'">
+    {{ timeLeft.hours }}
+    <span :class="unitClass || 'unit'"
+      >hour{{ timeLeft.hours !== 1 ? 's' : '' }}</span
+    >
+    {{ timeLeft.minutes }}
+    <span :class="unitClass || 'unit'"
+      >minute{{ timeLeft.minutes !== 1 ? 's' : '' }}</span
+    >
+  </span>
+  <span v-else-if="timeLeft.minutes > 0" :class="valueClass || 'value'">
+    {{ timeLeft.minutes }}
+    <span :class="unitClass || 'unit'"
+      >minute{{ timeLeft.minutes !== 1 ? 's' : '' }}</span
+    >
+  </span>
+  <span v-else-if="timeLeft.seconds > 0" :class="valueClass || 'value'">
+    {{ timeLeft.seconds }}
+    <span :class="unitClass || 'unit'"
+      >second{{ timeLeft.seconds !== 1 ? 's' : '' }}</span
```

Comment:
Lots of redundancy on the plural checks here - thoughts on using something like pluralize? https://www.npmjs.com/package/pluralize

Reply:
Pluralize docs:

> For applications where the word(s) are known ahead of time, you can use a simple ternary (or function) which would be a much lighter alternative.

Figured I'd stick with just extracting this into a reusable ternary function